web/src/views/DandisetLandingView/DandisetDetails.vue

```diff
@@ -152,18 +154,102 @@ export default {
 
       return null;
     },
-    ...mapState('girder', {
-      currentDandiset: (state) => state.currentDandiset,
+    currentVersion() {
+      if (!this.versions) { return draftVersion; }
+      return this.versions[this.currentVersionIndex].version;
+    },
+    currentDandiset() {
+      // Done this way because we'll want to add in
+      // fetching stats from the publish endpoint later on.
+      return this.girderDandiset;
+    },
+    currentVersionIndex() {
+      if (!this.publishDandiset || !this.versions) {
+        return 0;
+      }
+
+      const index = this.versions.findIndex(
+        ({ version }) => this.publishDandiset.version === version,
+      );
+
+      return index === -1 ? 0 : index;
+    },
+    ...mapState('dandiset', {
+      girderDandiset: (state) => state.girderDandiset,
+      publishDandiset: (state) => state.publishDandiset,
     }),
   },
+  asyncComputed: {
+    async versions() {
+      const { identifier } = this.girderDandiset.meta.dandiset;
+
+      try {
+        const { results } = await publishRest.versions(identifier);
+        return [
+          // First entry is null as it represents the draft dandiset
+          { version: draftVersion },
+          ...results,
+        ];
+      } catch (err) {
+        return [];
+      }
+    },
+  },
+  watch: {
+    currentVersion: {
+      immediate: true,
+      handler(version) {
+        this.setRouteVersion(version);
+      },
+    },
+  },
   methods: {
+    validDandisetVersion,
+    setVersion(index) {
```

Comment:
I don't see this method called anywhere. Is it used?

Reply:
If you're referring to `validDandisetVersion`: 

https://github.com/dandi/dandiarchive/blob/8f5a1489549416cd516c8b84d3f4ded4b08c16f4/web/src/views/DandisetLandingView/DandisetDetails.vue#L64

If you're referring to `setVersion`:

https://github.com/dandi/dandiarchive/blob/8f5a1489549416cd516c8b84d3f4ded4b08c16f4/web/src/views/DandisetLandingView/DandisetDetails.vue#L108